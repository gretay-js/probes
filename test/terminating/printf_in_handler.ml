let () =
  [%probe "boo" (Printf.printf "BOO!\n")];
  let boo_enabled = [%probe_is_enabled "boo"] in
  if boo_enabled then Printf.printf "BOO was enabled - first top level\n"

let f x y z =
  Printf.printf "f: %d %f %s\n" x y z

let foo a b c =
  [%probe "fooia" (f a b c)];
  Printf.printf "from foo %d %f\n" a b

let rec fib i j =
  if (i >= 0) then begin
    if (i mod 2 = 0) then
      foo (i+j) Float.((of_int i) /. (of_int j)) "myau";
    fib j (i+j)
  end

let () =
  [%probe "boo" (Printf.printf "BOO!\n")];
  fib 0 1;
  let boo_enabled = [%probe_is_enabled "boo"] in
  if boo_enabled then Printf.printf "BOO was enabled - end\n";
