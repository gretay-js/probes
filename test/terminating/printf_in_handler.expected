BOO!
BOO was enabled - first top level
BOO!
f: 1 0.000000 myau
from foo 1 0.000000
f: 5 0.666667 myau
from foo 5 0.666667
f: 21 0.615385 myau
from foo 21 0.615385
f: 89 0.618182 myau
from foo 89 0.618182
f: 377 0.618026 myau
from foo 377 0.618026
f: 1597 0.618034 myau
from foo 1597 0.618034
f: 6765 0.618034 myau
from foo 6765 0.618034
f: 28657 0.618034 myau
from foo 28657 0.618034
f: 121393 0.618034 myau
from foo 121393 0.618034
f: 514229 0.618034 myau
from foo 514229 0.618034
f: 2178309 0.618034 myau
from foo 2178309 0.618034
f: 9227465 0.618034 myau
from foo 9227465 0.618034
f: 39088169 0.618034 myau
from foo 39088169 0.618034
f: 165580141 0.618034 myau
from foo 165580141 0.618034
f: 701408733 0.618034 myau
from foo 701408733 0.618034
f: 2971215073 0.618034 myau
from foo 2971215073 0.618034
f: 12586269025 0.618034 myau
from foo 12586269025 0.618034
f: 53316291173 0.618034 myau
from foo 53316291173 0.618034
f: 225851433717 0.618034 myau
from foo 225851433717 0.618034
f: 956722026041 0.618034 myau
from foo 956722026041 0.618034
f: 4052739537881 0.618034 myau
from foo 4052739537881 0.618034
f: 17167680177565 0.618034 myau
from foo 17167680177565 0.618034
f: 72723460248141 0.618034 myau
from foo 72723460248141 0.618034
f: 308061521170129 0.618034 myau
from foo 308061521170129 0.618034
f: 1304969544928657 0.618034 myau
from foo 1304969544928657 0.618034
f: 5527939700884757 0.618034 myau
from foo 5527939700884757 0.618034
f: 23416728348467685 0.618034 myau
from foo 23416728348467685 0.618034
f: 99194853094755497 0.618034 myau
from foo 99194853094755497 0.618034
f: 420196140727489673 0.618034 myau
from foo 420196140727489673 0.618034
f: 1779979416004714189 0.618034 myau
from foo 1779979416004714189 0.618034
f: -1683258232108429379 -0.631133 myau
from foo -1683258232108429379 -0.631133
BOO was enabled - end
